src/Utils/obiDialogParser.ts

```diff
@@ -0,0 +1,331 @@
+                            extractorStep: { textVariations: [] },
+                            scorerSteps: []
+                        }
+                    }
+                    if (!step.activity) {
+                        throw new Error("Expected activity to be set")
+                    }
+                    const scorerStep = await this.getScorerStepfromActivity(step.activity)
+                    trainRound.scorerSteps.push(scorerStep)
+                }
+                else if (step.$type === "Microsoft.TextInput") {
+                    if (!trainRound) {
+                        trainRound = {
+                            extractorStep: { textVariations: [] },
+                            scorerSteps: []
+                        }
+                    }
+                    if (!step.prompt) {
+                        throw new Error("Expected activity to be set")
+                    }
+                    const scorerStep = await this.getScorerStepfromActivity(step.prompt)
+                    trainRound.scorerSteps.push(scorerStep)
+                }
+                else if (step.$type === "Microsoft.BeginDialog") {
+                    const subDialog =  this.composerDialog.dialogs.find(d => d.$id === step.dialog)
+                    if (!subDialog) {
+                        throw new Error(`Dialog name ${step.dialog} undefined`)
+                    }
+
+                    const childDialogs = await this.getTrainDialogsfromOBIDialog(subDialog)
+                    
+                    // Add children to train dialog list
+                    trainDialogs = [...trainDialogs, ...childDialogs]
+                }
+                else if (step.$type !== "Microsoft.EndTurn") {
+                    console.log(`Unhandled OBI Type: ${step.$type}`)
+                }
+            }
+            if (trainRound) {
+                if (trainDialogs.length === 0) {
+                    trainDialogs.push(this.makeEmptyTrainDialog())
+                }
+                for (const td of trainDialogs) {
+                    // If susequent round has no extractor step, just prepend scorer steps
+                    if (td.rounds.length > 0 && td.rounds[0].extractorStep.textVariations.length === 0) {
+                        td.rounds[0].scorerSteps = [...trainRound.scorerSteps, ...td.rounds[0].scorerSteps]
+                    }
+                    // Otherwise prepend round
+                    else {
+                        td.rounds = [trainRound, ...td.rounds]
+                    }
+                }
+            }
+        }
+        return trainDialogs
+    }
+    
+    private async getScorerStepfromActivity(prompt: string): Promise<CLM.TrainScorerStep> {
+    
+        const parsedActivity = prompt.substring(prompt.indexOf("[") + 1, prompt.lastIndexOf("]")).trim()
+        let response = this.composerDialog.lgMap.get(parsedActivity)
+        if (!response) {
+            // LARS temp to handle badly create dialog
```

Comment:
Suggest rephrasing as a concrete TODO

(eg, `// TODO : throw Error once parsing is implemented for all activity types`)

...and removing the commented-out throw below